src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+        }
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (11,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //         field = null; // 1
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(11, 9),
+                // (12,23): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //         _ = new C() { field = null }; // 2
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(12, 23),
+                // (25,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //             field = null; // 3
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(25, 13),
+                // (30,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //             field = null; // 4
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(30, 13),
+                // (38,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //         field = null; // 5
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(38, 9),
+                // (42,9): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //         field = null; // 6
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(42, 9),
+                // (48,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the class in which the field is defined or a variable initializer))
+                //             field = null; // 7
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(48, 13)
+                );
+        }
+
+        [Fact]
+        public void StaticReadonlyFieldInitializedByAnother()
+        {
+            string source = @"
+public class C
+{
+    public static readonly int field;
+    public static readonly int field2 = (field = 42);
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
```

Comment:
>comp.VerifyDiagnostics(); [](start = 12, length = 25)

We want to run success cases and observe expected behavior, this is a general observation for all tests added in this file.

Reply:
I don't think all tests need execution validation, given the nature of the change. Picked a sample.

---
In reply to: [415209699](https://github.com/dotnet/roslyn/pull/43275#discussion_r415209699) [](ancestors = 415209699)

Reply:
> I don't think all tests need execution validation, given the nature of the change. Picked a sample.

Let's sync up offline

---
In reply to: [415508812](https://github.com/dotnet/roslyn/pull/43275#discussion_r415508812) [](ancestors = 415508812,415209699)